File: services/src/main/java/com/liferay/mobile/sdk/v7/flagsentry/FlagsEntryService.java
package com.liferay.mobile.sdk.v7.flagsentry;

import com.liferay.mobile.sdk.Call;
import com.liferay.mobile.sdk.annotation.Param;
import com.liferay.mobile.sdk.annotation.Path;
import com.liferay.mobile.sdk.http.Response;
import java.lang.String;
import org.json.JSONObject;

@Path("/flags.flagsentry")
public interface FlagsEntryService {
  @Path("/add-entry")
  Call<Response> addEntry(@Param(name = "className") String className, @Param(name = "classPK") long classPK, @Param(name = "reporterEmailAddress") String reporterEmailAddress, @Param(name = "reportedUserId") long reportedUserId, @Param(name = "contentTitle") String contentTitle, @Param(name = "contentURL") String contentURL, @Param(name = "reason") String reason, @Param(name = "serviceContext", className = "com.liferay.portal.kernel.service.ServiceContext") JSONObject serviceContext);
}

File: windows/Liferay.SDK/Service/V62/FlagsEntry/FlagsEntryService.cs
//------------------------------------------------------------------------------
// <copyright file="FlagsEntryService.cs">
//    Copyright (c) 2014-present Andrea Di Giorgi
//
//    Permission is hereby granted, free of charge, to any person obtaining a
//    copy of this software and associated documentation files (the "Software"),
//    to deal in the Software without restriction, including without limitation
//    the rights to use, copy, modify, merge, publish, distribute, sublicense,
//    and/or sell copies of the Software, and to permit persons to whom the
//    Software is furnished to do so, subject to the following conditions:
//
//    The above copyright notice and this permission notice shall be included in
//    all copies or substantial portions of the Software.
//
//    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
//    IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
//    FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
//    THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
//    LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
//    FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
//    DEALINGS IN THE SOFTWARE.
// </copyright>
// <author>Andrea Di Giorgi</author>
// <website>https://github.com/Ithildir/liferay-sdk-builder-windows</website>
//------------------------------------------------------------------------------

using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Threading.Tasks;

namespace Liferay.SDK.Service.V62.FlagsEntry
{
	public class FlagsEntryService : ServiceBase
	{
		public FlagsEntryService(ISession session)
			: base(session)
		{
		}

		public async Task AddEntryAsync(string className, long classPK, string reporterEmailAddress, long reportedUserId, string contentTitle, string contentURL, string reason, JsonObjectWrapper serviceContext)
		{
			var _parameters = new JsonObject();

			_parameters.Add("className", className);
			_parameters.Add("classPK", classPK);
			_parameters.Add("reporterEmailAddress", reporterEmailAddress);
			_parameters.Add("reportedUserId", reportedUserId);
			_parameters.Add("contentTitle", contentTitle);
			_parameters.Add("contentURL", contentURL);
			_parameters.Add("reason", reason);
			this.MangleWrapper(_parameters, "serviceContext", "com.liferay.portal.service.ServiceContext", serviceContext);

			var _command = new JsonObject()
			{
				{ "/flagsentry/add-entry", _parameters }
			};

			await this.Session.InvokeAsync(_command);
		}
	}
}